HLTriggerOffline/Scouting/plugins/PatElectronTagProbeAnalyzer.cc

```diff
@@ -0,0 +1,843 @@
+
+  histos.hTrack_qoverpModeError_Barrel =
+      ibook.book1D(name + "_Track_qoverpModeError_Barrel", name + "_Track_qoverpModeError_Barrel", 36, -4.5, 4.5);
+  histos.hTrack_qoverpModeError_Endcap =
+      ibook.book1D(name + "_Track_qoverpModeError_Endcap", name + "_Track_qoverpModeError_Endcap", 36, -4.5, 4.5);
+
+  histos.hRelEcalIsolation_Barrel =
+      ibook.book1D(name + "_RelEcalIsolation_Barrel", name + "_RelEcalIsolation_Barrel", 100, 0, 1.0);
+  histos.hRelEcalIsolation_Endcap =
+      ibook.book1D(name + "_RelEcalIsolation_Endcap", name + "_RelEcalIsolation_Endcap", 100, 0, 1.0);
+
+  histos.hRelHcalIsolation_Barrel =
+      ibook.book1D(name + "_RelHcalIsolation_Barrel", name + "_RelHcalIsolation_Barrel", 100, 0, 1.0);
+  histos.hRelHcalIsolation_Endcap =
+      ibook.book1D(name + "_RelHcalIsolation_Endcap", name + "_RelHcalIsolation_Endcap", 100, 0, 1.0);
+
+  histos.hRelTrackIsolation_Barrel =
+      ibook.book1D(name + "_RelTrackIsolation_Barrel", name + "_RelTrackIsolation_Barrel", 100, 0, 1.0);
+  histos.hRelTrackIsolation_Endcap =
+      ibook.book1D(name + "_RelTrackIsolation_Endcap", name + "_RelTrackIsolation_Endcap", 100, 0, 1.0);
+
+  histos.hPtvsInvMass_Barrel =
+      ibook.book2D(name + "_PtvsInvMass_Barrel", name + "_PtvsInvMass_Barrel", 5, 0, 100, 10, 80, 100);
+  histos.hPtvsInvMass_Endcap =
+      ibook.book2D(name + "_PtvsInvMass_Endcap", name + "_PtvsInvMass_Endcap", 5, 0, 100, 10, 80, 100);
+
+  histos.hEtavsInvMass = ibook.book2D(name + "_EtavsInvMass", name + "_EtavsInvMass", 10, -2.5, 2.5, 10, 80, 100);
+
+  histos.hInvMass = ibook.book1D(name + "_Invariant_Mass", name + "_Invariant_Mass", 800, 0, 200);
+}
+
+// ------------ method fills 'descriptions' with the allowed parameters for the
+// module  ------------
+void PatElectronTagProbeAnalyzer::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  // The following says we do not know what parameters are allowed so do no
+  // validation Please change this to state exactly what you do use, even if it
+  // is no parameters
+  edm::ParameterSetDescription desc;
+  desc.add<std::string>("OutputInternalPath", "MY_FOLDER");
+  desc.add<edm::InputTag>("TriggerResultTag", edm::InputTag("TriggerResults", "", "HLT"));
+  desc.add<edm::InputTag>("TriggerObjects", edm::InputTag("slimmedPatTrigger"));
+  desc.add<edm::InputTag>("ElectronCollection", edm::InputTag("slimmedElectrons"));
+  desc.add<edm::InputTag>("ScoutingElectronCollection", edm::InputTag("Run3ScoutingElectrons"));
+  desc.add<edm::InputTag>("eleIdMapTight",
+                          edm::InputTag("egmGsfElectronIDs:cutBasedElectronID-RunIIIWinter22-V1-tight"));
+  descriptions.add("PatElectronTagProbeAnalyzer", desc);
+}
+
+double PatElectronTagProbeAnalyzer::getPtFromEnergyMassEta(double energy, double mass, double eta) const {
+  double theta = 2.0 * std::atan(std::exp(-eta));
+  double pt = std::sqrt(energy * energy - mass * mass) * std::sin(theta);
+  return pt;
+}
```

Comment:
```suggestion
double PatElectronTagProbeAnalyzer::getPtFromEnergyMassEta(double energy, double mass, double eta) const {
    double transverseEnergy = std::sqrt(energy * energy - mass * mass);
    return transverseEnergy / std::cosh(eta);
}
```

the way it is looks inefficient.